### PR TITLE
Calculate variation between current and prev period

### PR DESCRIPTION
**Pull request by dpordomingo**
It will be called the api doubling the passed DateInterval.
Once the response is obtained, it will be calculated the variation between the previous period, and the passed DateInterval.
Once obtained the variation, it's returned the data only for the passed DateInterval.

### CODE REVIEW

File: src/js/services/api/index.js [hunk 1]
```diff
@@ -67,83 +98,82 @@ export const getContributors = (token, userAccount, from, to, repos) => {
...
-          if (!thumbsData[metricNo]) {
-            thumbsData[metricNo] = [];
-          }
-
-          thumbsData[metricNo].push({ x: new Date(day.date), y: dateTime.secondsToHours(value) });
-        });
-      });
+export const getMetrics = (api, accountId, dateInterval, repos, contributors) => {
+  const metrics = ['lead-time', 'wip-time', 'review-time', 'merging-time', 'release-time'];
+  const result = metrics.reduce((acc, metric) => {
```
Comment (se7entyse7en) [inline src/js/services/api/index.js]: No need to reduce here, you're not aggregating you're just doing initialization.

File: src/js/services/api/index.js [hunk 2]
```diff
@@ -67,83 +98,82 @@ export const getContributors = (token, userAccount, from, to, repos) => {
...
+  };
 
+  return fetchApiMetricsLine(api, metrics, accountId, doubleDateInterval, repos, contributors).then(apiData => {
+    if (!apiData.calculated || !apiData.calculated[0]) {
+      throw new Error('API returned no calculated data');
     }
 
-    return getPipelineThumbs(thumbsData);
+    // This will unfold values, to have separate data per different metric, instead of all metrics under the same day.
+    apiData.calculated[0].values.forEach(step => {
```
Comment (se7entyse7en) [inline src/js/services/api/index.js]: Could you remind me when this happens `len(calculated) > 1`?

File: src/js/services/api/index.js [hunk 3]
```diff
@@ -67,83 +98,82 @@ export const getContributors = (token, userAccount, from, to, repos) => {
...
+      step.values.forEach((value, metricNo) => {
+        if (new Date(step.date) >= dateInterval.from) {
```
Comment (se7entyse7en) [inline src/js/services/api/index.js]: The date can be initialized once outside the `forEach`.

**Comment by se7entyse7en** [comment]
@dpordomingo I reviewed only the commit that is part of this PR as you said in the PR desc.

File: src/js/components/ui/Badge.jsx [hunk 4]
```diff
@@ -7,25 +7,29 @@ import { faAngleDown, faAngleUp, faEquals } from '@fortawesome/free-solid-svg-ic
 export default ({ value, trend, className }) => {
   const commonClasses = ['badge', 'font-weight-normal', 'align-middle', 'd-inline-block'];
   let customClasses, icon;
+  let suffix = '%';
   if (!trend) {
     customClasses = ['badge-pill', 'badge-secondary', 'py-1', 'px-2'];
   } else if (value < 0) {
     customClasses = ['badge-danger'];
+    value = -value;
```
[the remark this answers is not in the document]
Comment (dpordomingo) [reply src/js/components/ui/Badge.jsx]: more clear at least!.

File: src/js/services/api/index.js (hunk 1 above)
Comment (dpordomingo) [reply src/js/services/api/index.js]: I must recognize that I use `reduce` too much, and you found a way to reduce to absurdity the way I use it that much.

File: src/js/services/api/index.js (hunk 2 above)
Comment (dpordomingo) [reply src/js/services/api/index.js]: The API returns as much `calculated` items, as `for` filters you pass the `/metrics_line` endpoint.
The webapp is currently using only one as you can see in [`api/fetchApiMetricsLine` method sources](https://github.com/athenianco/athenian-webapp/blob/master/src/js/services/api/index.js#L140)

Once we refactor the API as we have in our wishlist, that method will be also affected, in order to reinforce that logic.

File: src/js/services/api/index.js (hunk 3 above)
Comment (dpordomingo) [reply src/js/services/api/index.js]: optimizing cpu cycles! that's neat! thanks!

**Comment by dpordomingo** [comment]
Thanks for revieweing @se7entyse7en !
I addressed your suggestions by 7eca21696766322fbb0b44392fc66977ed629562

**Pull request merged**